recipe_recommender.mdx

```diff
@@ -0,0 +1,395 @@
+retrieve a list of recipe file paths.
+
+<CodeGroup>
+```python io_utils.py
+import os
+
+from settings import DATA_DIR
+
+def get_recipe_filepaths():
+    results = []
+
+    for file in os.listdir(DATA_DIR):
+        if file.endswith('.txt'):
+            results.append(DATA_DIR / file)
+
+    return results
+```
+</CodeGroup>
+
+Next, create a `files.py` file that contains the functionality used to parse file names and read text files. The
+file name (without the extension) acts as the unique ID for each recipe vector stored in Pinecone.
+
+<CodeGroup>
+```python files.py
+import os
+
+def extract_filename_without_extension(path):
+    """Extracts the filename without its extension from a given path"""
+    base_name = os.path.basename(path)
+    file_name_without_extension, _ = os.path.splitext(base_name)
+    return file_name_without_extension
+
+def read_txt_file(filepath):
+    """Reads the content of a text file and returns it as a string"""
+    with open(filepath, 'r') as file:
+        content = file.read()
+
+    return content
+```
+</CodeGroup>
+
+Create a `pinecone_populate.py` file next. Once created, run the script. During execution, the script first creates a
+Pinecone index. It then reads each of our recipes, generates a vector from it, and then inserts the vector as a new
+record in our Pinecone index.
+
+<CodeGroup>
+```python pinecone_populate.py
+import os
+
+import pinecone
+
+from files import extract_filename_without_extension, read_txt_file
+from io_utils import get_recipe_filepaths
+from openai_utils import get_vector_for_text
+from settings import PINECONE_INDEX_NAME
+
+PINECONE_API_KEY = os.environ.get('PINECONE_API_KEY')
+
+def create_index_if_not_exists():
+    """Create the given index if it does not already exist"""
+    indexes = pinecone.list_indexes()
+
+    if PINECONE_INDEX_NAME not in indexes:
+        pinecone.create_index(PINECONE_INDEX_NAME, dimension=1024, metric='euclidean')
+
+def main():
+    create_index_if_not_exists()
+    index = pinecone.Index(PINECONE_INDEX_NAME)
+
+    for filepath in get_recipe_filepaths():
+        recipe_id = extract_filename_without_extension(filepath)
+        recipe_content = read_txt_file(filepath)
+        vector = get_vector_for_text(recipe_content)
+        index.upsert([(recipe_id, vector)])
+
+if __name__ == '__main__':
+    pinecone.init(api_key=PINECONE_API_KEY, environment='gcp-starter')
+    main()
+```
+</CodeGroup>
+
+# Building the Recommender
+
+Before we query our index, let's first create a `suggestion.py` file. This logic allows us to create a prompt which
+is text that includes:
+1. an indication of food we are interested in (eg. "I like chicken")
+2. a list of recipes based on the query results from Pinecone
+
+<CodeGroup>
+```python suggestion.py
+import os
+
+import promptlayer
+
+openai = promptlayer.openai
+openai.api_key = os.getenv('OPENAI_API_KEY')
+
+def add_recipes_to_prompt(prompt, recipes_list):
```

Comment:
This feels like the right time to introduce PromptLayer prompt templates & use it for this

Reply:
Maybe start the section off creating a prompt template in promptlayer for the search query

Reply:
Let's explain more about the prompt! This is the meat&potatoes of this example. We are a prompt engineering platform this is the good stuff